src/FixPressureAtNodeAlgorithm.C

```diff
@@ -56,35 +56,79 @@ FixPressureAtNodeAlgorithm::execute()
   if (doInit_)
     initialize();
 
-  // Reset LHS and RHS for this matrix
-  eqSystem_->linsys_->resetRows(refNodeList_, 0, 1);
-
   int numNodes = refNodeList_.size();
   ThrowAssertMsg(numNodes <= 1,
                  "Invalid number of nodes encountered in FixPressureAtNodeAlgorithm");
 
-  // Fix the pressure for this node only if this is proc is owner
-  if (numNodes > 0 && fixPressureNode_) {
-    std::vector<double> lhs(1);
-    std::vector<double> rhs(1);
-    std::vector<int> scratchIds(1);
-    std::vector<double> scratchVals(1);
-    stk::mesh::Entity node = refNodeList_[0];
-    const double pressureN = *stk::mesh::field_data(*pressure_, node);
+  stk::mesh::Entity targetNode = targetNode_;
+  if (numNodes == 0 || !realm_.bulk_data().is_valid(targetNode)) {
+    return;
+  }
 
-    lhs[0] = 1.0; // Set diagonal entry to 1.0
-    rhs[0] = info_.refPressure_ - pressureN;
+  // Reset LHS and RHS for this matrix
+  CoeffApplier* coeffApplier = eqSystem_->linsys_->get_coeff_applier();
+#ifdef KOKKOS_ENABLE_CUDA
+  CoeffApplier* deviceCoeffApplier = coeffApplier->device_pointer();
+#endif
+ 
+  ngp::Mesh ngpMesh = realm_.ngp_mesh();
+  NGPDoubleFieldType ngpPressure = realm_.ngp_field_manager().get_field<double>(pressure_->mesh_meta_data_ordinal());
+  double refPressure = info_.refPressure_;
+  const bool fixPressureNode = fixPressureNode_;
+
+  const int bytes_per_team = 0;
+  const int rhsSize = 1;
+  const int bytes_per_thread = rhsSize*((rhsSize+1)*sizeof(double)+2*sizeof(int)) +
+                               sizeof(SharedMemView<double**,DeviceShmem>) +
+                               sizeof(SharedMemView<double*,DeviceShmem>) +
+                               2*sizeof(SharedMemView<int*,DeviceShmem>);
+
+  const int threads_per_team = 1;
+  auto team_exec = get_device_team_policy(1, bytes_per_team, bytes_per_thread, threads_per_team);
+
+  Kokkos::parallel_for(team_exec, KOKKOS_LAMBDA(const DeviceTeamHandleType& team)
+  {
+    SharedMemView<double**,DeviceShmem> lhs = get_shmem_view_2D<double,DeviceTeamHandleType,DeviceShmem>(team, rhsSize, rhsSize);
+    SharedMemView<double*,DeviceShmem> rhs = get_shmem_view_1D<double,DeviceTeamHandleType,DeviceShmem>(team, rhsSize);
+    SharedMemView<int*,DeviceShmem> scratchIds = get_shmem_view_1D<int,DeviceTeamHandleType,DeviceShmem>(team, rhsSize);
+    SharedMemView<int*,DeviceShmem> sortPerm = get_shmem_view_1D<int,DeviceTeamHandleType,DeviceShmem>(team, rhsSize);
+
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, 1), [=](const size_t& )
+    {
+      stk::mesh::Entity deviceTargetNode = targetNode;
```

Comment:
Is there a GPU quirk that requires a second variable even though `targetNode` is already in device scope?

Reply:
This can probably be cleaned up. I was having trouble getting things working, and I think other issues were really at fault. I'll do a clean-up commit.